Assets/Python/UniquePowers.py

```diff
@@ -260,71 +233,61 @@ def seljukUP(self, city):
 		
 
 	def arabianUP(self, city):
-		pArabia = gc.getPlayer(iArabia)
+		#pArabia = gc.getPlayer(iArabia)
 		iStateReligion = pArabia.getStateReligion()
 
-		if (iStateReligion >= 0):
-			#for iReligion in range(iNumReligions):	# Leoreth: now removes foreign religions and buildings (except holy cities) as well
-			#	if city.isHasReligion(iReligion) and not city.isHolyCityByType(iReligion):
-			#		city.setHasReligion(iReligion, False, False, False)
-			#	if city.hasBuilding(iTemple + iReligion*4):
-			#		city.setHasRealBuilding((iTemple + iReligion*4), False)
-			#	if city.hasBuilding(iCathedral + iReligion*4):
-			#		city.setHasRealBuilding((iCathedral + iReligion*4), False)
-			#	if city.hasBuilding(iMonastery + iReligion*4):
-			#		city.setHasRealBuilding((iMonastery + iReligion*4), False)
-
-			if (not city.isHasReligion(iStateReligion)):
+		if iStateReligion >= 0:
+			if not city.isHasReligion(iStateReligion):
 				city.setHasReligion(iStateReligion, True, True, False)
-			if (not city.hasBuilding(iTemple + iStateReligion*4)):
+			if not city.hasBuilding(iTemple + iStateReligion*4):
 				city.setHasRealBuilding((iTemple + iStateReligion*4), True)
 
 #------------------AZTEC U.P.-------------------
 
 	def aztecUP(self, argsList): #Real Slavery by Sevo
-		pWinningUnit,pLosingUnit = argsList
+		if not pAztecs.isAlive(): return
+		if utils.isReborn(iAztecs): return
+		
+		pWinningUnit, pLosingUnit = argsList
 		pWinningPlayer = gc.getPlayer(pWinningUnit.getOwner())
 		
-		if (pWinningPlayer.getID() != iAztecs):
+		if pWinningPlayer.getID() != iAztecs:
 			return
 			
-		if utils.isReborn(iAztecs): return
-
 		pLosingPlayer = gc.getPlayer(pLosingUnit.getOwner())
 		cLosingUnit = PyHelpers.PyInfo.UnitInfo(pLosingUnit.getUnitType())
 		
-		if (pLosingUnit.getUnitType() < iWarrior):
+		if pLosingUnit.getUnitType() < iWarrior:

```

Comment:
I think this is supposed to exclude animals, so it'd be better to use `pLosingUnit.isAnimal()`.